mllib/src/main/scala/org/apache/spark/ml/tuning/TrainValidationSplit.scala

```diff
@@ -161,11 +174,28 @@ class TrainValidationSplit @Since("1.5.0") (@Since("1.5.0") override val uid: St
     }
 
     // Wait for all metrics to be calculated
-    val metrics = metricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
-
-    // Unpersist training & validation set once all metrics have been produced
-    trainingDataset.unpersist()
-    validationDataset.unpersist()
+    val metrics = try {
+      metricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
+    }
+    catch {
+      case e: Throwable =>
+        subTaskFailed = true
+        try {
+          Thread.sleep(1000)
+          val sparkContext = sparkSession.sparkContext
+          sparkContext.cancelJobGroup(tvsJobGroup)
+        } catch {
+          case _: Throwable => ()
```

Comment:
Nit: is () necessary versus just nothing here?

Reply:
I think this is a standard way representing return "Unit", https://stackoverflow.com/questions/53415664/how-do-i-explicitly-return-unit